runtime/core/decoder/torch_asr_decoder.cc

```diff
@@ -220,6 +219,9 @@ void TorchAsrDecoder::UpdateResult() {
 void TorchAsrDecoder::AttentionRescoring() {
   searcher_->FinalizeSearch();
   UpdateResult();
+  if (0.0 == opts_.rescoring_weight) {
```

Comment:
we'd better not compare two float-pointing value, fabs(a - b) < 1e-6 would be safer

Reply:
Here we means exact zero, not close to 0. Actually we write in this way since we see a lot of code like this in kaldi and other open source c++ project.